### PR TITLE
Fix pybind11 problems with c10::SymInt unregistered

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #88011

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Review by weiwangmeta — APPROVED** [review]
Thanks for the fix!

**Comment by ezyang** [comment]
Testing diff (not linked) at https://www.internalfb.com/diff/D40817266

File: torch/csrc/utils/python_arg_parser.h [hunk 1]
```diff
@@ -79,82 +79,6 @@
 #include <string>
 #include <vector>
 
-namespace pybind11 {
-namespace detail {
-template <>
-struct type_caster<c10::SymInt> {
- public:
-  PYBIND11_TYPE_CASTER(c10::SymInt, _("SymInt"));
```
Comment (albanD) [inline torch/csrc/utils/python_arg_parser.h:87]: Wait the lint didn't catch that??

**Review by albanD — APPROVED** [review]

File: torch/csrc/utils/python_arg_parser.h (hunk 1 above)
Comment (ezyang) [reply torch/csrc/utils/python_arg_parser.h:87]: PYBIND11_SPECIALIZATION lint only checks for PYBIND11_DECLARE_HOLDER_TYPE. It is kind of annoying to make it apply to type caster as well

Comment (ezyang) [reply torch/csrc/utils/python_arg_parser.h:87]: also, the old caster was in this file and didn't cause any problems. Very perplexing.

**Pull request closed without merging**